Headers/Public/VLCTranscoder.h

```diff
@@ -0,0 +1,65 @@
+/*****************************************************************************
+ * VLCTranscoder.h: VLCKit.framework VLCTranscoder implementation
+ *****************************************************************************
+ * Copyright (C) 2018 Carola Nitz
+ * Copyright (C) 2018 VLC authors and VideoLAN
+ * $Id$
+ *
+ * Authors:  Carola Nitz <caro # videolan.org>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
+ *****************************************************************************/
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+* Transcoder delegate allows to be notified about transcoding state
+*/
+@protocol VLCTranscoderDelegate <NSObject>
+
+@optional
+/**
+ * Sent by the default notification center whenever the transcoder state has changed.
+ * \param aNotification
+ */
+- (void)transcoderFinishedSucessfully:(BOOL)success;
+
+@end
+
+/**
+ * Provides an object to convert a subtitle file and moviefile into one.
+ */
+@interface VLCTranscoder: NSObject
+
+/**
+ * the delegate object implementing the optional protocol
+ */
+@property (weak, nonatomic) id<VLCTranscoderDelegate> delegate;
+
+/**
+ * mux srt and mp4 file to an mkv file with embedded subtitles either with VideoToolbox-based H264 encoding or VP80 is Videotoolbox is not available
+ * \param srtPath path to srt file
+ * \param mp4Path path to mp4 file
+ * \param outPath path where the new file should be written to
+ * \return an BOOL with the success status, returns NO if the subtitle file is not an srt or mp4File is not an mp4 file or the files don't exist at that path or transcoding failed for other reasons
+ */
+
+- (BOOL)muxSubtitleFile:(NSString *)srtPath toMp4File:(NSString *)mp4Path outputPath:(NSString *)outPath;
```

Comment:
Method name implies muxing happens but in reality it re-encodes and remuxes, as far as I understand the code, which is a very important difference so should be clarified. (Muxing is relatively fast, encoding not)

Nitpick: `toMP4File` instead of `toMp4File` and probably `muxSRTFile` if it only works for SRT.

Reply:
all good points!

Reply:
`blendSubtitleFile` or `blendSubtitleFileAndSave` might be a better name, as the subtitle is not muxed but burned into the video.

It should work with any subtitle file for which the decoder is available

Reply:
@alexandre-janniaux the problem is I don't know exactly which subtitle files are supported by the decoder. Because I would like to state that explicitly

Reply:
okay I'd say this is something we can add in an update later. Lets go for now with srt and get this shipped!